### PR TITLE
[wip] Gpu rebase inprog

### PR DESCRIPTION
**Pull request by aeyakovenko**
need some help with the borrow checker :)

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fn main() {
```
Comment (garious) [inline build.rs]: What happens if the user doesn't have cuda libs installed or doesn't want to install them?

File: src/event.rs [hunk 2]
```diff
@@ -47,7 +47,7 @@ impl Event {
     /// spending plan is valid.
     pub fn verify(&self) -> bool {
         match *self {
-            Event::Transaction(ref tr) => tr.verify(),
+            Event::Transaction(ref tr) => tr.plan_verify(),
```
Comment (garious) [inline src/event.rs]: You should not have needed to make this change. I took verify() out of skel's codepath.

File: src/bin/testnode.rs [hunk 3]
```diff
@@ -11,7 +11,7 @@ use std::sync::atomic::AtomicBool;
 use std::sync::{Arc, Mutex};
 
 fn main() {
-    let addr = "127.0.0.1:8000";
+    let addr = "127.0.0.1:9000";
```
Comment (garious) [inline src/bin/testnode.rs]: No thanks!

File: src/transaction.rs [hunk 4]
```diff
@@ -8,12 +8,17 @@ use rayon::prelude::*;
 use signature::{KeyPair, KeyPairUtil, PublicKey, Signature, SignatureUtil};
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
-pub struct Transaction {
+pub struct Signed {
```
Comment (garious) [inline src/transaction.rs]: Last review asked that you change this to TransactionData and that you remove 'from' from it.

**Comment by garious** [comment]
ugh, a separate PR for a rebase? This throws away all the comments I made on the previous one.

File: build.rs (hunk 1 above)
Comment (sakridge) [reply build.rs]: Seems like we can have something like: `if env::var("CARGO_CFG_cuda").unwrap() { prints... }`

if we have a cfg like this:
`#[cfg(cuda)] => CARGO_CFG_cuda`

**Pull request closed without merging**